script.js

```diff
@@ -1,4 +1,704 @@
+//*blackjack!
+// Flow of the game
+// 1) Deck is shuffled.
+// 2) User clicks Submit to deal cards.
+// 3) The cards are analysed for game winning conditions, e.g. Blackjack.
+// 4) The cards are displayed to the user.
+// 5) The user decides whether to hit or stand, using the submit button to submit their choice.
+// 6) The user's cards are analysed for winning or losing conditions.
+// 7) The computer decides to hit or stand automatically based on game rules.
+// 8) The game either ends or continues.
+
+// ******* Step 1 : make card game (deck / shuffle / draw cards / get winners) *******//
+//      1) define player and dealer
+//      2) create and shuffle a game deck
+//      3) draw 2 cards for player and dealer each
+//      4) win (black jack / higher hand value)
+//      5) display and for both plaer and get who is the winner
+
+// ******** Step 2 : Player hit or stand ******* //
+//      1) extra game mode to hit or stand
+//      2) function to add card card after being input of "hit" or "stand"
+
+// ******** Step 3 : Dear hit or stand ***** //
+//      1) dealer choose to hit or stand after player finish their turn
+//      2) if dealer hand value is less than 17, hit
+//      3) if dealer hand value is more than 17, stand
+
+// ********* step 4 : Ace to be come either 1 or 11 ********* //
+//      1) if the total value goes below 21, Ace will be counted as 11
+//      2) if the total value goes above 21, Ace need to change value to 1 instead of 11
+
+//image global variable//
+var image1 =
+  '<img src="https://c.tenor.com/FaMwfY6UU9kAAAAM/shuffle-cards.gif"/>';
+var imageThumbsUp =
+  '<img src="https://c.tenor.com/D5d3KKgydBUAAAAM/gavin-meme.gif"/>';
+var imageWhat =
+  '<img src="https://c.tenor.com/8W03lgoMEv0AAAAM/umm-what.gif"/>';
+var imageEvilLaugh =
+  '<img src="https://c.tenor.com/Jw3KI0JGQFAAAAAM/gavin-gavin-thomas.gif"/>';
+var imageLost =
+  '<img src="https://c.tenor.com/9-0P-Fh_VSMAAAAM/gavin-meme.gif"/>';
+var imageThinking =
+  '<img src="https://c.tenor.com/BMQTJ4nWvCIAAAAM/thinking-think.gif"/>';
+var imageWrongAnswer =
+  '<img src="https://c.tenor.com/S9A24f_TI_EAAAAM/did-you-read-it-you-get-it.gif"/>';
+
+// *** GLOBAL VARIABLE ***//
+var gameMode_Start = "Game Start";
+var gameMode_DrawnCards = "Drawn Cards";
+var gameMode_ResultShown = "Show Result";
+var gameMode_HitOrStand = "Hit or Stand";
+var currentGameMode = gameMode_Start;
+
+// variable for the group of multiple cards holding
+var playerHand = [];
+var dealerHand = [];
+
+// *** FUNCTIONS *** //
+var checkForBlackJack = function (handArray) {
+  // this will return true or false!
+  var playerCardOne = handArray[0];
+  var playerCardTwo = handArray[1];
+  var isBlackJack = false;
+
+  //way to become blackjack (21)
+  //1) ACE + 10 or more  (must have ACE in either one)
+  //2) 10 or more + ACE
+  if (
```

Comment:
You rarely need a variable that stores true/ false directly. If you see this, you can probably write it a better way. In this case, just return line 70 and 71 directly.
```js
return (playerCardOne.name == "ace" && playerCardTwo.rank >= 10) || (playerCardOne.rank >= 10 && playerCardTwo.name == "ace")
```